### ISSUE TITLE
Narrator screen reader is not announcing any information about "Generating Previews" popup: A11y_.NET CoreWinforms_PrintingTesting_ScreenReader

### ISSUE THREAD

**Issue by PoojaNamde** [dotnet/winforms#11456]
### .NET version

dotnet-sdk-9.0.100-preview.5.24258.1

### Did it work in .NET Framework?

Yes

### GitHub Tags:
#A11yMAS; #A11yTCS; #A11ySev2; #BM_.NET Core[Winforms]_Win32_May2024; #.NETCore; #WCAG1.3.1; #DesktopApp; #Win11; #FTP; #Narrator; #NVDA; #Element:Dialog;

### Environment Details:
App name: .NET Core Winforms
Window Version: Win 11
OS Build: 22621.3085
Screen readers: Narrator, NVDA 2023.3.3

### Repro Steps:

1. Turn on Narrator/NVDA.
2. Launch the "Accessibility core app".
3. Main Form screen would appear.
4. TAB to "Printing_Controls:Testing the controls under Printing Tab" and hit ENTER key.
5. "Generating Previews" popup will appear.
6. Observe that Narrator screen reader is not announcing any information about "Generating Previews" popup.

### Actual Result:
Narrator screen reader is not announcing any information about "Generating Previews" popup

**Similar issue also observed with NVDA.**

### Expected Result:
Narrator screen reader should be announcing information about "Generating Previews" popup

### User Impact:
Screen reader users will not be able to get information about the popup.

### Attachment

[11456_A11y_.NET CoreWinforms_PrintingTesting_NVDA.webm](https://github.com/dotnet/winforms/assets/90901519/af87a892-0013-44a4-bba7-d7babc26b45f)

[11456_A11y_.NET CoreWinforms_PrintingTesting_Narrator.webm](https://github.com/dotnet/winforms/assets/90901519/87da83af-b0c4-4ec8-9d00-7f93f88fd14f)

**Comment by Philip-Wang01** [comment]
Same as https://github.com/dotnet/winforms/issues/9280. However, on .NET 9.0, the results are the same for NVDA and Narrator, and the screen reader does not announce any information about “Generating Previews”.

**Comment by Yash14j** [comment]
GithubTags:#Rev:yaja;

**Comment by merriemcgaw** [comment]
@LeafShi1 can you confirm whether this is our API with the popup dialog or if this comes from Windows.

**Comment by LeafShi1** [comment]
> can you confirm whether this is our API with the popup dialog or if this comes from Windows

This is our API , we will fix this issue

**Comment by Liv-Goh** [comment]
Verified this issue on the latest .NET 9.0.100-preview 6.24320.9, this issue still repro.

Narrator:
![narrator issue](https://github.com/dotnet/winforms/assets/169347065/7dbabf70-8753-4deb-a7ea-fec1d0da0a8a)

NVDA:
![nvda issue](https://github.com/dotnet/winforms/assets/169347065/27e48afb-7d73-4360-9fec-90e63d8d79ad)

**Comment by LeafShi1** [comment]
It's strange is that opening Narrator after the "Generating Previews" showing, the dialog can get focus and the content can be announced normally. If opening Narrator first and then debugging the project, the dialog "Generating Previews" cannot get focus.

**Comment by merriemcgaw** [comment]
Can I get an update on this issue? Will we have a resolution, even partial, for RC1?

**Comment by LeafShi1** [comment]
> @LeafShi1 Can I get an update on this issue? Will we have a resolution, even partial, for RC1?

Currently this problem is blocked, because Narrator cannot recognize the [new thread](https://github.com/LeafShi1/winforms/blob/6243ed5a5222c476bf45e41c9d17aff73b3056b9/src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintControllerWithStatusDialog.BackgroundThread.cs#L23)

**Comment by JeremyKuhne** [comment]
@merriemcgaw please follow up on this one

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - had you seen that narrator can't focus on dialogs that start on helper threads in in native applications - 

`This problem is because StatusDialog runs in a new thread, and Narrator cannot focus on the new thread. NVDA can recognize the new thread normally.`

**Comment by merriemcgaw** [comment]
@Olina-Zhang can your team check to see if JAWS responds correctly in this scenario? I'm going to bring this to Office Hours to see if we can close it or they'd like to file a Narrator bug. It would be nice to know if we're good on both free screen readers when I do.

**Comment by Olina-Zhang** [comment]
@merriemcgaw I tested JAWS and NVDA can responds and announce the label and cancel button in "Generating Previews" dialog correctly, but Narrator cannot.

**Comment by Epica3055** [comment]
@Tanya-Solyanik 

![issues11456](https://github.com/user-attachments/assets/4fb74e54-2ab1-45ff-8747-51d2e4ec68d5)

**Comment by PoojaNamde** [comment]
#Regressed:08-09-24;

Verified this issue in the latest build. 
1. Issue is still reproducing with Narrator as it is not announcing anything about popup.
2. NVDA is announcing about cancel button after loading 12 documents.
3. Issue is not reproducing with JAWS. Please find all the attachments for reference.

[JAWS attachment.webm](https://github.com/user-attachments/assets/6b180e34-5f1f-4e35-90ca-7057f0c91289)

[Narrator attachment.webm](https://github.com/user-attachments/assets/b71fb316-e361-4600-b3c2-b1702b79e17d)

[NVDA attachment.webm](https://github.com/user-attachments/assets/69a992dd-eeb5-43cb-85df-cc49981c6a66)

**Comment by Tanya-Solyanik** [comment]
@Epica3055 
>In native applications, narrator can't focus on dialogs that start on helper threads

Your screen short shows a winforms application. When we talk about native applications we usually mean MFC.

**Comment by merriemcgaw** [comment]
@Olina-Zhang please attach a picture of the UIA properties from AccessibilityInsights when that dialog is up to show it's got everything coded right and they'll close it.

**Comment by Olina-Zhang** [comment]
@merriemcgaw here is a screenshot of "Generating Previews" pane in AccessibilityInsights:
![image](https://github.com/user-attachments/assets/3b4b4d76-c224-45a0-95a9-c67beecc8934)

**Comment by PoojaNamde** [comment]
@Olina-Zhang Could you please share screenshot Accessibility insights including whole details. I tried to get the whole details but not able to inspect the control with AI tool.

**Comment by Olina-Zhang** [comment]
@PoojaNamde What's details you want to see? All properties of "Generating Previews" pane in AI tools?

**Comment by PoojaNamde** [comment]
@Olina-Zhang As we are able to see the details till 'Legacy accessible pattern.child id', we need whatever details present below it.

**Comment by Olina-Zhang** [comment]
@PoojaNamde It seems it hard to use AccessibilityInsights to focus on Generating Previews window, here is I captured:
![image](https://github.com/user-attachments/assets/167bc789-85f0-4283-a694-920488200929)

**Comment by merriemcgaw** [comment]
@PoojaNamde can you close this now?

**Comment by merriemcgaw** [comment]
@LeafShi1 @Olina-Zhang I spoke with the SMEs and we need to know whether this dialog is content that we own or if we're getting this from Windows APIs?

If we get this dialog from Windows we will apply an existing exception which is already documented about Windows Printing APIs. If we own the pop up window then we need to update the window's AccessibleName property and that will solve this issue. If we own this code, is it Sample app code or WinForms code? We'd want to make sure we get this in for GA if it is WinForms code.

**Comment by PoojaNamde** [comment]
#A11YCAIOH;  #CEACCReviewed;

As per AI4D Screenshots, Is Dialog property is false and Name property is not given. Is Dialog should be true and Name property should be defined. Screen reader should read the title of the dialog to close this bug and not necessary to read the content as it is for few seconds.

**Comment by LeafShi1** [comment]
@merriemcgaw  We own this pop-up window, and I tried to modify the `AccessibleName ` property of the window, but it didn't work. Narrator still can't focus on the dialog.

and the problem is not reading the dialog title or content, but Narrator cannot focus on the dialog, even if an Accessibility object is created for the dialog (which we have done in PR #11523 ). The root cause is that the dialog runs on a helper thread.

And I found that not only the StatusDialog we defined, but also the MessageBox running in the helper thread cannot be focused by Narrator

![BeforeChanges](https://github.com/user-attachments/assets/af173777-8451-4ae4-a8ca-554701805761)

**Comment by merriemcgaw** [comment]
If you set `IsDialog` and `AccessibleName` appropriately will at least NVDA and JAWS be able to read the title of the pop up window? If we can show we have those set correctly I think we will get permission to close this bug.

**Comment by LeafShi1** [comment]
> If you set `IsDialog` and `AccessibleName` appropriately will at least NVDA and JAWS be able to read the title of the pop up window? If we can show we have those set correctly I think we will get permission to close this bug.

Currently NVDA can read the Title of the pop-up window.
![AfterChanges](https://github.com/user-attachments/assets/86511a97-c460-4881-903a-e446077647dc)

**Comment by merriemcgaw** [comment]
@LeafShi1 Excellent!

Can we go ahead and set `IsDialog` property to make it `true` and ensure there is an `AccessibleName` property? If we do both of these we will be able to resolve this issue, regardless of what Narrator does. We will be doing the right thing.

**Comment by LeafShi1** [comment]
> @LeafShi1 Excellent!
> 
> Can we go ahead and set `IsDialog` property to make it `true` and ensure there is an `AccessibleName` property? If we do both of these we will be able to resolve this issue, regardless of what Narrator does. We will be doing the right thing.

The [current code](https://github.com/LeafShi1/winforms/blob/d876f24a18c1ad63690d19ef1e58dc1d8f798d4a/src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintControllerWithStatusDialog.StatusDialog.cs#L98) only creates an Accessible object for StatusDialog without setting IsDialog and AccessibleName, but NVDA and JAWS can read the Title Bar. Do we still need to set these two properties?
![AfterChanges](https://github.com/user-attachments/assets/bf3c7978-2c43-46af-9c89-a3b47e148975)

**Comment by LeafShi1** [comment]
@merriemcgaw The properties `IsDialog` and `AccessibleName` have been updated, can we close this issue?

**Comment by merriemcgaw** [comment]
Please show an image if possible of Accessibility Insights or Inspect that shows that so we can close this issue. @PoojaNamde be on the lookout for the image

**Comment by LeafShi1** [comment]
> Please show an image if possible of Accessibility Insights or Inspect that shows that so we can close this issue. @PoojaNamde be on the lookout for the image

The properties in Inspect:
<img width="835" alt="image" src="https://github.com/user-attachments/assets/6c227697-f81f-431f-bea4-ecc8f6aa09b3">

**Comment by PoojaNamde** [comment]
@LeafShi1 Could you please upload the latest build so that I can reverify the issue in latest build.

**Comment by LeafShi1** [comment]
> @LeafShi1 Could you please upload the latest build so that I can reverify the issue in latest build.

I can't provide you with a new build at the moment. We have to wait until the PR change flows into dotnet/sdk. You can verify this issue with the 10.0 SDK.

**Comment by lonitra** [comment]
Closing as this seems fixed based off of Leaf's comment. Please feel free to reopen if this is still an issue.

**Comment by Tanya-Solyanik** [comment]
@lonitra  - this bug should be verified and then closed by @PoojaNamde

**Comment by PoojaNamde** [comment]
@LeafShi1 I have followed the steps mentioned in private build. After that I have downloaded the "WinformsAccessibilityNET9App" and installed the winform app and verified the issue. Issue is still reproducing there.

Please correct me if I have followed the incorrect steps and help me on how to get new environement.

**Comment by LeafShi1** [comment]
> @LeafShi1 I have followed the steps mentioned in private build. After that I have downloaded the "WinformsAccessibilityNET9App" and installed the winform app and verified the issue. Issue is still reproducing there.
> 
> Please correct me if I have followed the incorrect steps and help me on how to get new environement.

I updated the WinformsAccessibilityNET9App, please verify again

**Comment by PoojaNamde** [comment]
@LeafShi1 Verified the issue in latest app, issue is still reproducing as screen reader is not announcing about the popup.


![image](https://github.com/user-attachments/assets/218e7c9e-95ad-41df-9ae1-99afd7d693a5)

**Comment by LeafShi1** [comment]
> @LeafShi1 Verified the issue in latest app, issue is still reproducing as screen reader is not announcing about the popup.
> 

For this fix, we only need to confirm that the IsDislog and AccessibleName properties are correct. Have you verified it in Inspect or Accessibility Insights?

<img width="545" alt="image" src="https://github.com/user-attachments/assets/f7b11330-1731-4178-9851-1d27d6106e55">

**Comment by PoojaNamde** [comment]
#Closed; #Regressed:10-01-24;

As per the above comment, IsDislog and AccessibleName properties are correct and JAWS is reading the correct information. Closing this bug.
